server/jetstream_cluster.go

```diff
@@ -6067,6 +6067,53 @@ func (mset *stream) checkClusterInfo(ci *ClusterInfo) {
 	}
 }
 
+// Return a list of alternates, ranked by preference order to the request, of stream mirrors.
+// This allows clients to select or get more information about read replicas that could be a
+// better option to connect to versus the original source.
+func (js *jetStream) streamAlternates(ci *ClientInfo, stream string) []StreamAlternate {
+	js.mu.RLock()
```

Comment:
Apparently, you need to check for `js != nil` here. In the code above where you invoke this, the `js.clusterInfo()` is immune to nil pointer because it checks. So either call js.streamAlternates when js is not nil, or check for nil here.